service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -264,6 +252,55 @@ private String prepareNarrativeStatement(Observation observation, String comment
         return TemplateUtils.fillTemplate(NARRATIVE_STATEMENT_TEMPLATE, narrativeStatementTemplateParameters.build());
     }
 
+    private String prepareStatementsForDerivedObservations(IdMapper idMapper, List<Observation> derivedObservations) {
+        StringBuilder derivedObservationsBlock = new StringBuilder();
+
+        derivedObservations.forEach(derivedObservation -> {
+            var observationStatement = prepareObservationStatement(idMapper, derivedObservation);
+            var narrativeStatement = prepareNarrativeStatement(idMapper, derivedObservation);
+
+            if (!observationStatement.isEmpty() && !narrativeStatement.isEmpty()) {
+                var compoundStatementId = idMapper.getOrNew(ResourceType.Observation, derivedObservation.getIdElement());
+                var codeElement = prepareCodeElement(derivedObservation);
+                var classCode = prepareClassCode(derivedObservation);
+                var effectiveTime = StatementTimeMappingUtils.prepareEffectiveTimeForObservation(derivedObservation);
+                var availabilityTimeElement = StatementTimeMappingUtils.prepareAvailabilityTimeForObservation(derivedObservation);
+
+                var observationCompoundStatementTemplateParameters = ObservationCompoundStatementTemplateParameters.builder()
+                    .compoundStatementId(compoundStatementId)
+                    .classCode(classCode.getCode())
+                    .codeElement(codeElement)
+                    .effectiveTime(effectiveTime)
+                    .availabilityTimeElement(availabilityTimeElement)
+                    .observationStatement(observationStatement)
+                    .narrativeStatement(narrativeStatement);
+
+                derivedObservationsBlock.append(
+                    TemplateUtils.fillTemplate(
+                        OBSERVATION_COMPOUND_STATEMENT_TEMPLATE,
+                        observationCompoundStatementTemplateParameters.build()
+                    )
+                );
+            } else {
+                derivedObservationsBlock.append(observationStatement).append(narrativeStatement);
+            }
+        });
+
+        return derivedObservationsBlock.toString();
+    }
+
+    private CommentType prepareCommentType(Observation observation) {
+        if (observation.getComment().equals("EMPTY REPORT")) {
```

Comment:
Is a comment guaranteed to be present or shall we yoda it?
```suggestion
        if ("EMPTY REPORT".equals(observation.getComment())) {
```

Reply:
Good point, but we do a `observation.hasComment()` check beforehand so this is safe